ERCS/erc-7291.md

```diff
@@ -0,0 +1,696 @@
+  - For cases where an expiry time is not needed, the expiry time **SHOULD** be set to infinity.
+
+- This proposal defines a base specification of what a PBM should entail. Extensions to this base specification can be implemented as separate specifications.
+
+### PBM Architecture
+
+In this EIP, we propose a modular PBM architecture that has three distinct components (the core, the plugins and the hooks):
+
+- The **core components** contains basic functionalities and validation checks that all PBMs should have. Core components includes sovToken and PBM wrapper containing the core logic (e.g. logic to whitelist the merchant/redeemer address, logic to unwrap upon transfer to a whitelisted address, logic for minting and burning the PBM) and a token manager which allows for token registration, retrieval. In addition, the PBM wrapper **MAY** include logic to interface with plugins and hooks.
+- The **plugin components** implement additional functionality that only specific PBMs may require (e.g. logic to call external application programming interfaces to verify specific PBM condition was met, logic to track PBM usage patterns).
+- The **hook components** implement additional validation checks that some PBMs may need (e.g. checks for expiration, daily spending limit, goods & services limit etc).
+
+For example, a PBM creator may want to ensure only 50% of PBM Series A can be spent in supermarkets, while there are no restriction on proportion of PBM Series B that can be spent in supermarkets. He can implement a plugin smart contract to keep track of supermarket spendings by PBM users. He will also implement a hook to validate that less than 50% of the PBM Series A issued a user is spent in a supermarket. This allows him to use the same generic PBM wrapper and sovToken for both PBM Series A and B. In addition, PBM series A will register the plugin module and hook module for additional data tracking and validations.
+
+### Auditability
+
+PBM Wrapper **SHOULD** provide the public easily accessible mechanism(s) to verify the smart contract logic for unwrapping a PBM. Such mechanisms could be leveraged by automated validation or asynchronous user verifications from transacting parties and/or whitelisted third parties attestations.
+
+As the fulfilment of PBM conditions is likely to be subjected to audits to ensure trust amongst all transacting parties, the following evidence shall be documented to support audits:
+
+- The interface/events emitted **SHOULD** allow a fine-grained recreation of the transaction history, token types and token balances
+- The source code **SHOULD** be verified and formally published on a blockchain explorer.
+
+### Fungibility
+
+A PBM Wrapper **SHOULD** be able to wrap multiple types of compatible Spot Tokens (e.g. the same PBM Wrapper should be able to wrap USDC and XSGD). Spot Tokens wrapped by the same PBM wrapper may or may not be fungible to one another. The standard does NOT mandate how an implementation must do this.
```

Comment:
Is a Spot Token the same as a sovToken? I didn't see the term defined before here and I am unfamiliar with it.

Reply:
Yes, you are right, a Spot Token is the same as a sovToken. I have changed references to Spot Token to sovToken instead.